freqtrade/freqai/data_handler.py

```diff
@@ -0,0 +1,439 @@
+import json
+import os
+import copy
+import numpy as np
+import pandas as pd
+from pandas import DataFrame
+from joblib import dump
+from joblib import load
+from sklearn.model_selection import train_test_split
+from sklearn.metrics.pairwise import pairwise_distances
+import datetime
+from typing import Any, Dict, List, Tuple
+import pickle as pk
+from freqtrade.configuration import TimeRange
+
+SECONDS_IN_DAY = 86400
+
+class DataHandler:
```

Comment:
I would somehow rename this if possible (both the class and the file)
we have an `IDataHandler` (which is the base class for json/hdf5/... hanlders) - and having a (almost identical) name is quite confusing, especially considering that this does A LOT more.

This is more like  Model Preparation or data preparation than just a "datahandler" (it does PCA, and probably much more i didn't look at yet in detail...)

If we fail to come up with a better name, then maybe `ModelDataHandler` would already be better.

Reply:
Thanks for the comments, all great points - I will get to work.

Reply:
How about FreqaiDataHandler?

Reply:
it's a possibility (better than just DataHandler for sure).

i'm not particularly thrilled by the `DataHandler` part in the name, really (but lack a better term for now)

it is handling data for sure - but it's really preparing / handling the dataset for the model ... so maybe something around DataSet could also work (naming is hard - and also not a problem we need to solve now i think).

Reply:
FreqaiDataKitchen ;)?